src/IssueViz.UnitTests/Selection/AnalysisIssueNavigationTests.cs

```diff
@@ -0,0 +1,265 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2020 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Moq;
+using SonarLint.VisualStudio.Integration.UnitTests;
+using SonarLint.VisualStudio.IssueVisualization.Models;
+using SonarLint.VisualStudio.IssueVisualization.Selection;
+using SonarLint.VisualStudio.IssueVisualization.TableControls;
+
+namespace SonarLint.VisualStudio.IssueVisualization.UnitTests.Selection
+{
+    [TestClass]
+    public class AnalysisIssueNavigationTests
+    {
+        private Mock<IAnalysisIssueSelectionService> selectionServiceMock;
+
+        private AnalysisIssueNavigation testSubject;
+
+        [TestInitialize]
+        public void TesInitialize()
+        {
+            selectionServiceMock = new Mock<IAnalysisIssueSelectionService>();
+
+            testSubject = new AnalysisIssueNavigation(selectionServiceMock.Object);
+        }
+
+        [TestMethod]
+        public void MefCtor_CheckIsExported()
+        {
+            var selectionService = Mock.Of<IAnalysisIssueSelectionService>();
+            var selectionServiceExport = MefTestHelpers.CreateExport<IAnalysisIssueSelectionService>(selectionService);
+
+            MefTestHelpers.CheckTypeCanBeImported<AnalysisIssueNavigation, IAnalysisIssueNavigation>(null, new[] { selectionServiceExport });
+        }
+
+        public enum GotoDirection
+        {
+            Next,
+            Previous
+        }
+
+        [TestMethod]
+        [DataRow(GotoDirection.Next)]
+        [DataRow(GotoDirection.Previous)]
+        public void GotoLocation_NoCurrentFlow_NoNavigation(GotoDirection direction)
+        {
+            SetupCurrentFlow((IAnalysisIssueFlowVisualization)null);
+            SetupCurrentLocation(CreateLocation(1, true));
+
+            if (direction == GotoDirection.Next)
```

Comment:
Data-driven tests that have conditional behaviour based on the input always feel wrong to me - seems to go against the spirit of data-driven testing. Still, it doesn't make much difference in practice in this case.

Reply:
Yea, the IFs was more lines than the test itself. I split them to separate tests.